packages/addresses/src/program-derived-address.ts

```diff
@@ -45,14 +46,14 @@ export function assertIsProgramDerivedAddress<TAddress extends string = string>(
     const validFormat =
         Array.isArray(value) && value.length === 2 && typeof value[0] === 'string' && typeof value[1] === 'number';
     if (!validFormat) {
-        // TODO: Coded error.
-        throw new Error(
-            `Expected given program derived address to have the following format: [Address, ProgramDerivedAddressBump].`,
-        );
+        throw new SolanaError(SOLANA_ERROR__INVALID_SEEDS, {
+            reason: `Expected given program derived address to have the following format: [Address, ProgramDerivedAddressBump].`,
+        });
     }
     if (value[1] < 0 || value[1] > 255) {
-        // TODO: Coded error.
-        throw new Error(`Expected program derived address bump to be in the range [0, 255], got: ${value[1]}.`);
+        throw new SolanaError(SOLANA_ERROR__INVALID_SEEDS, {
+            reason: `Expected program derived address bump to be in the range [0, 255], got: ${value[1]}.`,
+        });
```

Comment:
This isn't true, right? This is rather that the `ProgramDerivedAddress` is malformed. We don't really know about the seeds – just that the `ProgramDerivedAddress` isn't of the form `[Address, Bump]`